packages/prettier-config/README.md

```diff
@@ -0,0 +1,76 @@
+# `@side/prettier-config`
+
+This is a [shared configuration][shared] for [Prettier][prettier] that is used within JavaScript, Node.js, and TypeScript projects across `reside-eng`.
+
+## Setup
+
+### Install Prettier with Shared Configuration
+
+Install Prettier and this shared configuration package.
+
+```bash
+yarn add -D prettier @side/prettier-config
+```
+
+Then, update package.json to instruct Prettier to use the shared configuration.
+
+```jsonc
+// package.json
+{
+  "prettier": "@side/prettier-config"
+}
+```
+
+### Configure ESLint
+
+If using ESLint, install Prettier's ESLint config.
```

Comment:
Can you make a note that this is only need if for some reason they are not using `@side/...` ESLint config? We should have this included in our ESLint base config so teams don't need to deal with extra config.

Reply:
It might be worth discussing this a bit further:

Regardless of if we keep Prettier's config within the base or not, all consumer repos will have to do the following no matter what:
- Install `prettier` as a dev dependency.
- Install `@side/prettier-config` as a dev dependency.
- Use a `.prettierrc` file or package.json to point Prettier to use the custom config.
- Add a script to the lint-staged configuration that instructs Prettier to format specified file extensions.
- ~~Add `prettier` as the last config in ESLint's `extends` array.~~

I think that removing the last step (~~strikethroughed~~) makes our Prettier setup harder to grasp, since it is the only step here that would be abstracted away into a different package. 

Aside from our goal to have ESLint contain a single "extends" package, are there other strong reasons why we should keep the Prettier ESLint config out of consumer repos?

Reply:
> Install @side/prettier-config as a dev dependency.

I'm questioning why we would need each team to manage the above. Unless they are pulling in another ESLint library that would conflict, our library should be able to take care of this for them. One less thing to setup and manage as a dependency.

Reply:
Hey @hershmire! – circling back to this after the holidays 🎄 

I think the TL;DR here is that ESLint is great for linting but is subpar at other tasks like formatting, which is why Prettier recommends not using `eslint-plugin-prettier`. 

If our goal is to remove configuration steps from individual repos, two options come to mind for doing that:
- Create our own CLI that wraps over Prettier, ESLint, and any other libraries you want to abstract.
- Use a monorepo (which would mean there is only one set of configuration).
 
Since both of those would be pretty costly, I think that adopting the `@side/prettier-config` approach will be our best option for the time being.

Reply:
Yes, apologies. I don't have an issue with `@side/prettier-config`. For some reason I thought that was referring to the `eslint-config-prettier` which [turns off all rules that are unnecessary or might conflict with Prettier](https://github.com/prettier/eslint-config-prettier). This is what I'm saying we should include in our library so each team doesn't need to manage.

Reply:
Gotcha! 

I think the only tricky thing with abstracting `eslint-config-prettier` into ESLint configs (as opposed to requiring it in each consumer repo) is that repos will become susceptible to having rules that conflict with Prettier. Here's how that would occur:

```jsonc
// .eslintrc
{
  "extends": [
    "@side/eslint-config-react",
    "some-one-off-config-for-a-plugin" // adds lint rules that Prettier breaks
  ]
}
```

This scenario will be prevented if `eslint-config-prettier` is included at the consumer level, which is instructed in the `@side/prettier-config` setup.

```jsonc
// .eslintrc
{
  "extends": [
    "@side/eslint-config-react",
    "some-one-off-config-for-a-plugin", // adds lint rules that Prettier breaks
   "prettier" // ensures Prettier rules take precedence
  ]
}
```

Reply:
@hershmire do you have an idea of how you would prevent this scenario if `eslint-config-prettier` was abstracted? As 

far as I can tell, the most reliable way to prevent this buggy scenario would be having consumer repos include `eslint-config-prettier` as the last config in `extends`, which is depicted in the second code block from the previous comment.

Reply:
Yeah, the scenario you're suggesting could come up but I would think that's very rare circumstances. I'd imagine it could come up more often if a team was to pull in another library in general. But even then, that might be a very rare scenario. And the team could just turn off those offending rules. What I'm after is minimizing the extra boilerplate for teams. 

I'm okay with moving forward as is for now since I see us using `eslint-config-prettier` with our (not yet created) lint config libs that are specifically tailored for our apps – i.e. `@side/eslint-config-next` for UI apps. At this point, my expectation for UI apps would be to just have this in their .eslintrc.js file:

```js
module.exports = {
  extends: '@side/eslint-config-next',
}
```

Reply:
Sounds good! I will add `eslint-config-prettier` back in to the other lint packages so that consumer repos do not need to add it. I will also update the README for the prettier-config package.